examples/quick_start/scripts/preproc/README.md

```diff
@@ -0,0 +1,250 @@
+  --to_int16            Cast these columns (comma-sep) to int16, to save some
+                        memory.
+  --to_int8             Cast these columns (comma-sep) to int32, to save some
+                        memory.
+  --to_float32 
+                        Cast these columns (comma-sep) to float32
+```
+
+# Filtering and frequency capping
+```
+  --categ_min_freq_capping
+                        Value used for min frequency capping. If greater than 0, all categorical values which are less frequent than this threshold will be mapped to the null value encoded id.
+                        
+  --min_user_freq 
+                        Users with frequency lower than this value are removed
+                        from the dataset (before data splitting).
+  --max_user_freq 
+                        Users with frequency higher than this value are
+                        removed from the dataset (before data splitting).
+  --min_item_freq 
+                        Items with frequency lower than this value are removed
+                        from the dataset (before data splitting).
+  --max_item_freq 
+                        Items with frequency higher than this value are
+                        removed from the dataset (before data splitting).
+  --num_max_rounds_filtering 
+                        Max number of rounds interleaving users and items
+                        frequency filtering. If a small number of rounds is
+                        chosen, some low-frequent users or items might be kept
+                        in the dataset. Default is 5
+  --filter_query 
+                        A filter query condition compatible with dask-cudf
+                        `DataFrame.query()`
+```
+
+### Dataset splitting (train and eval sets)
+```
+  --dataset_split_strategy {random,random_by_user,temporal}
+                        If None, no data split is performed. If 'random',
+                        samples are assigned randomly to eval set according to
+                        --random_split_eval_perc. If 'random_by_user', users
+                        will have examples in both train and eval set,
+                        according to the proportion specified in
+                        --random_split_eval_perc. If 'temporal', the
+                        --timestamp_feature with
+                        --dataset_split_temporal_timestamp to split eval set
+                        based on time.
+  --random_split_eval_perc 
+                        Percentage of examples to be assigned to eval set. It
+                        is used with --dataset_split_strategy 'random' and
+                        'random_by_user'
+  --dataset_split_temporal_timestamp 
+                        Used when --dataset_split_strategy 'temporal'. It
+                        assigns for eval set all examples where the
+                        --timestamp_feature >= value
+```
+
+### CUDA cluster options
```

Comment:
as we discussed, I think this should be optional, and users should be able to use CPU as well, which they can prefer with a small dataset. This is the case for the recsys23 competition dataset, it is not that big.

Reply:
I changed the preprocess.py script to detect if there are GPU available, otherwise it configures `Dataset(..., cpu=True)`. But when testing this setting in the Merlin TF container 23.02 without GPUs available it raised some [errors](https://nvidia.slack.com/archives/CUM5PBE9W/p1682025737440849) when importing NVTabular due to a known  issue with cuda-python 11.7.0 and earlier (used by cudf). According to @oliverholworthy, it seems we won't have this issue with the release `23.04` because we’re using a more recent version of cudf and cuda-python.

Reply:
In addition to that, I created an option `--enable_dask_cuda_cluster` in the preprocess.py script to enable/disable the usage of Dask Cluster, as for smaller datasets not using LocalCUDACluster might be faster